community/lms/doctype/lms_course/lms_course.py

```diff
@@ -131,15 +131,30 @@ def get_batches(self, mentor=None):
             memberships = frappe.db.get_all(
                 "LMS Batch Membership",
                 {"member": member},
-                ["name"], as_dict=1)
+                ["batch"])
             batch_names = {m.batch for m in memberships}
-            return [b for b in batches if b.name in batch_names]
+            mentor_batches =  [b for b in batches if b.name in batch_names]
+            return self.get_batch_mentors(mentor_batches)
 
     def get_upcoming_batches(self):
         now = frappe.utils.nowdate()
-        return find_all("LMS Batch",
+        batches =  find_all("LMS Batch",
             course=self.name,
             start_date=[">", now])
+        batches = self.get_batch_mentors(batches)
+        return batches
+
+    def get_batch_mentors(self, batches):
+        for batch in batches:
+            batch.mentors = []
+            mentors = frappe.get_all(
+                        "LMS Batch Membership",
+                        {"batch": batch.name, "member_type": "Mentor"},
+                        ["member"])
+            for mentor in mentors:
+                member = frappe.db.get_value("Community Member", mentor.member, ["full_name", "photo", "abbr"], as_dict=1)
+                batch.mentors.append(member)
+        return batches
```

Comment:
I think this should become `.get_mentors` method in `LMSBatch`. The templates can call `batch.get_mentors()` to get the mentors.